File: fix_a11y.js
const fs = require('fs');
const path = require('path');

function processDir(dir) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const fullPath = path.join(dir, file);
    const stat = fs.statSync(fullPath);
    if (stat.isDirectory()) {
      processDir(fullPath);
    } else if (file === 'page.tsx') {
      let content = fs.readFileSync(fullPath, 'utf8');
      
      // Skip builder/page.tsx as it already has main
      if (fullPath.includes('\\builder\\') || fullPath.includes('/builder/')) continue;
      
      let changed = false;
      
      // Fix sequential heading issue (h2 -> h4 to h2 -> h3)
      if (content.includes('<h4 className="text-xl')) {
        content = content.replace(/<h4 className="text-xl/g, '<h3 className="text-xl');
        content = content.replace(/<\/h4>/g, '</h3>');
        changed = true;
      }
      
      // Fix missing landmark (main) issue
      if (content.includes('<div className="min-h-screen bg-slate-50')) {
        content = content.replace('<div className="min-h-screen bg-slate-50', '<main className="min-h-screen bg-slate-50');
        content = content.replace(/<\/div>\s*\);\s*}/, '</main>\n  );\n}');
        changed = true;
      }
      
      if (changed) {
        fs.writeFileSync(fullPath, content);
        console.log(`Updated ${fullPath}`);
      }
    }
  }
}

processDir(path.join(__dirname, 'app'));

File: fix_a11y.py
import os

app_dir = r"c:\Users\mtayy\OneDrive\Desktop\PROJECT\schedule-builder\app"

for root, dirs, files in os.walk(app_dir):
    for file in files:
        if file == "page.tsx":
            filepath = os.path.join(root, file)
            if "\\builder\\" in filepath or "/builder/" in filepath:
                continue
            
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()
                
            changed = False
            
            if '<h4' in content:
                content = content.replace('<h4', '<h3').replace('</h4>', '</h3>')
                changed = True
                
            if '<div className="min-h-screen bg-slate-50' in content:
                content = content.replace('<div className="min-h-screen bg-slate-50', '<main className="min-h-screen bg-slate-50')
                
                # Replace the last </div> before ); }
                parts = content.rsplit('</div>', 1)
                if len(parts) == 2:
                    content = '</main>'.join(parts)
                changed = True
                
            if changed:
                with open(filepath, 'w', encoding='utf-8') as f:
                    f.write(content)
                print("Updated " + filepath)
